Test testStorageOpt fails for docker 1.12.1

## Description

When doing a maven test on docker 1.12.1 (via Travis), the same test fails on multiple Pull Requests including #602 and #604 

## How to reproduce

mvn clean test

## What do you expect

This test should pass. 

## What happened instead

Instead, the error states that aufs does not support --storage-opt parameters.
```
  testStorageOpt(com.spotify.docker.client.DefaultDockerClientTest): Request error: POST unix://localhost:80/containers/create?name=2c27144a8128f2e1-5600b1b4b777125a: 500, body: {"message":"--storage-opt is not supported for aufs"}(..)
```

## Software:

- `docker version`: 1.12.1
- Spotify's docker-client version: 7.0.3-SNAPSHOT

## Full backtrace

```text
testStorageOpt(com.spotify.docker.client.DefaultDockerClientTest)  Time elapsed: 0.475 sec  <<< ERROR!
com.spotify.docker.client.exceptions.DockerRequestException: Request error: POST unix://localhost:80/containers/create?name=2c27144a8128f2e1-5600b1b4b777125a: 500, body: {"message":"--storage-opt is not supported for aufs"}
	at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2122)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2073)
	at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:583)
	at com.spotify.docker.client.DefaultDockerClientTest.testStorageOpt(DefaultDockerClientTest.java:3996)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
	at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:1020)
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:816)
	at org.glassfish.jersey.client.JerseyInvocation.access$700(JerseyInvocation.java:92)
	at org.glassfish.jersey.client.JerseyInvocation$5.completed(JerseyInvocation.java:773)
	at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:198)
	at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:79)
	at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:180)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:340)
	at org.glassfish.jersey.client.ClientRuntime$3.run(ClientRuntime.java:210)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

--- a/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
+++ b/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
@@ -340,6 +340,11 @@ public class DefaultDockerClientTest {
     assumeTrue(msg, dockerApiVersionAtLeast(required));
   }
 
+  private void requireStorageDriverNotAufs() throws Exception {
+    Info info = sut.info();
+    assumeFalse(info.storageDriver().equals("aufs"));
+  }
+
   private boolean dockerApiVersionAtLeast(final String expected) throws Exception {
     return compareVersion(dockerApiVersion, expected) >= 0;
   }
@@ -408,11 +413,11 @@ public class DefaultDockerClientTest {
         buildParam
     );
   }
-  
+
   @Test
   public void testHealthCheck() throws Exception {
     requireDockerApiVersionAtLeast("1.24", "health check");
-    
+
     // Create image
     final String dockerDirectory = Resources.getResource("dockerDirectoryWithHealthCheck")
             .getPath();
@@ -420,13 +425,13 @@ public class DefaultDockerClientTest {
             Paths.get(dockerDirectory),
             "test-healthcheck"
             );
-   
+
     // Inpect image to check healthcheck configuration
     final ImageInfo imageInfo = sut.inspectImage(imageId);
     assertThat(imageInfo.config().healthcheck(), notNullValue());
     assertThat(imageInfo.config().healthcheck().get("Test"),
             is(Arrays.asList("CMD-SHELL", "exit 1")));
-    
+
     // Create container based on this image to check initial container health state
     final ContainerConfig config = ContainerConfig.builder()
             .image("test-healthcheck")
@@ -435,9 +440,9 @@ public class DefaultDockerClientTest {
     final ContainerCreation creation = sut.createContainer(config, name);
     sut.startContainer(creation.id());
     final ContainerInfo containerInfo = sut.inspectContainer(creation.id());
-    
+
     assertThat(containerInfo.state().health(), notNullValue());
-    assertThat(containerInfo.state().health().status(), is("starting")); 
+    assertThat(containerInfo.state().health().status(), is("starting"));
   }
 
   @SuppressWarnings("emptyCatchBlock")
@@ -1837,11 +1842,11 @@ public class DefaultDockerClientTest {
     assertThat(actual.dns(), equalTo(expected.dns()));
     assertThat(actual.cpuQuota(), equalTo(expected.cpuQuota()));
   }
-  
+
   @Test
   public void testUpdateContainer() throws Exception {
     requireDockerApiVersionAtLeast("1.22", "update container");
-    
+
     final String containerName = randomName();
     final HostConfig hostConfig = HostConfig.builder()
             .cpuShares(256L)
@@ -1853,19 +1858,19 @@ public class DefaultDockerClientTest {
 
     sut.pull(BUSYBOX_LATEST);
     final ContainerCreation container = sut.createContainer(config, containerName);
-    
+
     final ContainerInfo containerInfo = sut.inspectContainer(container.id());
     assertThat(containerInfo.hostConfig().cpuShares(), is(256L));
-    
+
     final HostConfig newHostConfig = HostConfig.builder()
             .cpuShares(512L)
             .build();
     final ContainerUpdate containerUpdate = sut.updateContainer(containerInfo.id(), newHostConfig);
-    
+
     assertThat(containerUpdate.warnings(), is(nullValue()));
-    
+
     final ContainerInfo newContainerInfo = sut.inspectContainer(container.id());
-    
+
     assertThat(newContainerInfo.hostConfig().cpuShares(), is(512L));
   }
 
@@ -3404,7 +3409,7 @@ public class DefaultDockerClientTest {
       assertThat(network.internal(), is(false));
       assertThat(network.enableIPv6(), is(false));
     }
-    
+
     sut.removeNetwork(network.id());
 
     exception.expect(NetworkNotFoundException.class);
@@ -3911,7 +3916,7 @@ public class DefaultDockerClientTest {
 
     assertThat(info.hostConfig().oomScoreAdj(), is(500));
   }
-  
+
   @Test
   public void testPidsLimit() throws Exception {
     if (OsUtils.isLinux()) {
@@ -3935,7 +3940,7 @@ public class DefaultDockerClientTest {
 
     assertThat(info.hostConfig().pidsLimit(), is(100));
   }
-  
+
   @Test
   public void testTmpfs() throws Exception {
     requireDockerApiVersionAtLeast("1.22", "TmpFs");
@@ -3944,7 +3949,7 @@ public class DefaultDockerClientTest {
     sut.pull(BUSYBOX_LATEST);
 
     final ImmutableMap<String, String> tmpfs = ImmutableMap.of("/tmp", "rw,noexec,nosuid,size=50m");
-    
+
     final ContainerConfig config = ContainerConfig.builder()
         .image(BUSYBOX_LATEST)
         .hostConfig(HostConfig.builder()
@@ -3975,23 +3980,23 @@ public class DefaultDockerClientTest {
 
     assertThat(info.hostConfig().readonlyRootfs(), is(true));
   }
-  
+
 
   @Test
   public void testStorageOpt() throws Exception {
     requireDockerApiVersionAtLeast("1.24", "StorageOpt");
-      
+    requireStorageDriverNotAufs();
     // Pull image
     sut.pull(BUSYBOX_LATEST);
 
     final ImmutableMap<String, String> storageOpt = ImmutableMap.of("size", "20G");
-    
+
     final ContainerConfig config = ContainerConfig.builder()
-        .image(BUSYBOX_LATEST)
-        .hostConfig(HostConfig.builder()
-                        .storageOpt(storageOpt)
-                        .build())
-        .build();
+                      .image(BUSYBOX_LATEST)
+                      .hostConfig(HostConfig.builder()
+                          .storageOpt(storageOpt)
+                          .build())
+                      .build();
 
     final ContainerCreation container = sut.createContainer(config, randomName());
     final ContainerInfo info = sut.inspectContainer(container.id());
@@ -4055,7 +4060,7 @@ public class DefaultDockerClientTest {
                     .name(networkName).build());
 
     final String networkId = networkCreation.id();
-    
+
     assertThat(networkId, is(notNullValue()));
 
     final TaskSpec taskSpec = TaskSpec.builder()

